### DISCUSSION TITLE
System prompt not working in Long Note QA and Vault QA

### DISCUSSION THREAD

**Discussion by devinat1** [logancyang/obsidian-copilot#545] [Q&A] [answered]
![image](https://github.com/logancyang/obsidian-copilot/assets/15788108/e99e93ea-f341-43c1-a9cf-b6f303909b7d)

**Describe how to reproduce**
Use the following system prompt:
```
Your role is to act as a teacher who provides brief, to-the-point responses. For every query, your goal is to offer concise and informative answers, limiting yourself to just one sentence. This approach aims to deliver clear, direct, and efficient guidance, suitable for users seeking quick insights or straightforward information. In your responses, avoid elaborations, examples, or extended explanations. Stay focused on delivering the core message in the most succinct way possible.
```

Notice that with Long Note QA, you will get very long answers. 
The same issue occurs in 

**Expected behavior**
It should give the response in one sentence, as is the case with Chat mode.

**Additional context**
SYSTEM INFO:
	Obsidian version: v1.5.12
	Installer version: v1.4.16
	Operating system: Windows 10 Home 10.0.26100
	Login status: logged in
	Catalyst license: none
	Insider build toggle: off
	Live preview: on
	Base theme: dark
	Community theme: Minimal v7.5.4
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 66
	Plugins enabled: 34
		1: Templater v2.2.3
		2: Tab Switcher v1.4.0
		3: Google Photos v1.6.5
		4: Dataview v0.5.66
		5: Editor Syntax Highlight v0.1.3
		6: Minimal Theme Settings v7.5.0
		7: Commander v0.5.1
		8: Canvas Mindmap v0.1.9
		9: Auto Link Title v1.5.3
		10: Admonition v10.2.0
		11: Simple CanvaSearch v0.1.0
		12: QuickAdd v1.8.1
		13: Excalidraw v2.1.3
		14: Todoist Sync v1.12.0
		15: Linter v1.23.2
		16: Readwise Official v2.0.1
		17: Enhancing Export v1.10.4
		18: Zotero Integration v3.1.7
		19: Paste URL into selection v1.7.0
		20: Note Refactor v1.8.2
		21: Canvas Filter v0.9.4
		22: Ghost Fade Focus v3.0.0
		23: WakaTime v1.0.0
		24: Quick Switcher++ v4.1.0
		25: Recent Files v1.3.10
		26: ExcaliBrain v0.2.15
		27: Projects v1.17.2
		28: Periodic Notes v0.0.17
		29: Homepage v3.8.0
		30: Calendar v1.5.10
		31: Longform v2.0.7
		32: BRAT v1.0.1
		33: Slides Extended v2.0.6
		34: Copilot v2.5.2

**Comment by logancyang** [comment, marked as answer]
Actually this is a bug, my bad. Fixed.
